source/modulo_new_core/include/modulo_new_core/communication/MessagePair.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "modulo_new_core/communication/MessagePairInterface.hpp"
+
+#include <std_msgs/msg/bool.hpp>
+#include <std_msgs/msg/float64.hpp>
+#include <std_msgs/msg/float64_multi_array.hpp>
+#include <std_msgs/msg/int64.hpp>
+#include <std_msgs/msg/string.hpp>
+
+namespace modulo_new_core::communication {
+
+template<typename MsgT, typename DataT>
+class MessagePair : public MessagePairInterface {
+private:
+  std::shared_ptr<DataT> data_;
+
+public:
+  explicit MessagePair(std::shared_ptr<DataT> data);
+
+  [[nodiscard]] MsgT get_message() const;
```

Comment:
Thinking ahead to the needs of a subscriber, the name of this method should maybe be adjusted.

`get_message` -> `write_message` (creates a message from the data, internally calls `translators::write_msg`)
`read_message(MsgT)` (updates the data from a message, internally calls `translators::read_msg`)

```suggestion
  [[nodiscard]] MsgT write_message() const;
```